docs/conf.py

```diff
@@ -156,7 +156,19 @@ def check_dot_command():
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
 # The documentation build tool overrides this when extensions are included in the documentation build.
-exclude_patterns = ['adr/*.md']
+
+# Make sure iso-only documents are excluded from the doc build by default and only included for iso builds
+
+default_pattern = ['adr/*.md']
+exclude_from_oss = [
+    "lsm",
+    "administrators/operational_procedures_with_lsm.rst",
+    "administrators/support.rst"
+]
+exclude_patterns = default_pattern + exclude_from_oss
+
+if tags.has("iso"):
+    exclude_patterns = default_pattern
```

Comment:
I think we need to add a comment here to indicate why this cannot be achieved using the `only` directive.

Reply:
I've been doing some reading on this so I'll add some links that I found interesting:
- [docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-only) state "This directive is designed to control only content of document. It could not control sections, labels and so on."
- sphinx-doc/sphinx#9819
- sphinx-doc/sphinx#10642